samples/AudioEcsSample/main.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include <NovelRT/NovelRT.h>
+#include <memory>
+
+using namespace NovelRT::Ecs;
+using namespace NovelRT::Ecs::Audio;
+using namespace NovelRT::Input;
+using namespace NovelRT::PluginManagement;
+
+NovelRT::Utilities::Event<NovelRT::Timing::Timestamp> DummyUpdateStuff;
+
+int main()
+{
+    NovelRT::LoggingService logger = NovelRT::LoggingService();
+
+#if NOVELRT_MOLTENVK_VENDORED
+    auto icdPath = NovelRT::Utilities::Misc::getExecutablePath() / "MoltenVK_icd.json";
+    setenv("VK_ICD_FILENAMES", icdPath.c_str(), 0);
+    logger.logInfo("macOS detected - setting VK_ICD_FILENAMES to path: {}", icdPath.c_str());
+#endif
```

Comment:
We should probably open an issue about this too tbh.

Reply:
This is one and the same with the above issue, if I understand your concern

Reply:
#479